File: backend/src/models/SocialMedia.ts
import mongoose, {Schema, Types} from "mongoose";

export type SocialMedia = {
    influencer: Types.ObjectId;
    platform: "Instagram" | "YouTube" | "TikTok" | "Twitter" | "Facebook" | "LinkedIn";
    url: string;
    followersCount?: number;
    createdAt: Date;
}

const socialMediaSchema = new mongoose.Schema<SocialMedia>(
    {
        influencer: {
            type: Schema.Types.ObjectId,
            ref: "Influencer",
            required: [true, "Influencer is required"],
        },
        platform: {
            type: String,
            enum: ["Instagram", "YouTube", "TikTok", "Twitter", "Facebook", "LinkedIn"],
            required: [true, "Platform is required"],
        },
        url: {
            type: String,
            required: [true, "URL is required"],
            trim: true,
            match: [/^(https?:\/\/)?([\w\-])+\.{1}([a-zA-Z]{2,63})([\/\w\-.]*)*\/?$/, "Invalid URL"]
        },
        followersCount: {
            type: Number,
            default: 0,
            min: [0, "Followers count cannot be negative"],
        },
    },
    {
        timestamps: true,
    }
)

export const SocialMediaModel = mongoose.model<SocialMedia>("SocialMedia", socialMediaSchema);

File: backend/src/models/social_media.rs
use serde::{Serialize, Deserialize};
use mongodb::bson::{oid::ObjectId, DateTime};

#[derive(Debug, Serialize, Deserialize)]
pub struct SocialMedia {
    #[serde(rename = "_id", skip_serializing_if = "Option::is_none")]
    pub id: Option<ObjectId>,

    pub influencer: ObjectId,

    pub platform: String, // "Instagram" | "YouTube" | "TikTok" | "Twitter" | "Facebook"

    pub url: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub followers_count: Option<i64>,

    #[serde(rename = "createdAt", skip_serializing_if = "Option::is_none")]
    pub created_at: Option<DateTime>,
}
